src/VASSAL/counters/AreaOfEffect.java

```diff
@@ -199,14 +199,14 @@ public void draw(Graphics g, int x, int y, Component obs, double zoom) {
           AlphaComposite.SRC_OVER, transparencyLevel));
 
         Area a = getArea();
-
-        if (zoom != 1.0) {
-          a = new Area(AffineTransform.getScaleInstance(zoom,zoom)
-                                      .createTransformedShape(a));
+        
+        if (a != null) {
+          if (zoom != 1.0) {
+            a = new Area(AffineTransform.getScaleInstance(zoom,zoom).createTransformedShape(a));
+          }
+          g2d.fill(a);
         }
 
-        g2d.fill(a);
-
```

Comment:
Suggestion: If `a` is null, then nothing else in the same scope as it needs to be done. You could do this instead:
```
        Area a = getArea();
        if (a != null) {
          final Graphics2D g2d = (Graphics2D) g;

          final Color oldColor = g2d.getColor();
          g2d.setColor(transparencyColor);

          final Composite oldComposite = g2d.getComposite();
          g2d.setComposite(AlphaComposite.getInstance(
            AlphaComposite.SRC_OVER, transparencyLevel));

          if (zoom != 1.0) {
            a = new Area(AffineTransform.getScaleInstance(zoom,zoom)
                                        .createTransformedShape(a));
          }
  
          g2d.fill(a);
  
          g2d.setColor(oldColor);
          g2d.setComposite(oldComposite);
        }
```

Reply:
Good point. Changes Incorporated